opm/autodiff/WellConnectionAuxiliaryModule.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+  Copyright 2017 Dr. Blatt - HPC-Simulation-Software & Services
+  Copyright 2017 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WELLCONNECTIONAUXILIARYMODULE_HEADER_INCLUDED
+#define OPM_WELLCONNECTIONAUXILIARYMODULE_HEADER_INCLUDED
+
+#include <ewoms/aux/baseauxiliarymodule.hh>
+
+#include <opm/grid/CpGrid.hpp>
+
+#include <opm/core/wells.h>
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+
+namespace Opm
+{
+template<class TypeTag>
+class WellConnectionAuxiliaryModule
+    : public Ewoms::BaseAuxiliaryModule<TypeTag>
+{
+    typedef typename GET_PROP_TYPE(TypeTag, GlobalEqVector) GlobalEqVector;
+    typedef typename GET_PROP_TYPE(TypeTag, JacobianMatrix) JacobianMatrix;
+
+public:
+
+    using NeighborSet = typename
+        Ewoms::BaseAuxiliaryModule<TypeTag>::NeighborSet;
+
+    WellConnectionAuxiliaryModule(const Schedule& schedule,
+                                  const Dune::CpGrid& grid)
+    {
+        // Create cartesian to compressed mapping
```

Comment:
I'm pretty sure there are quite a few places this is already implemented, but I don't remember where on the fly. It would be good to consolidate on a single one.